File: src/main/java/it/contrader/model/Folder.java
package it.contrader.model;

import java.util.List;

import javax.persistence.Entity;
import javax.persistence.GeneratedValue;
import javax.persistence.GenerationType;
import javax.persistence.Id;
import javax.persistence.JoinColumn;
import javax.persistence.ManyToOne;
import javax.persistence.OneToMany;
import javax.persistence.OneToOne;
import javax.persistence.Table;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@AllArgsConstructor
@NoArgsConstructor
@Entity
public class Folder {
	
	@Id
	@GeneratedValue(strategy = GenerationType.IDENTITY)
	private Long id;
	
	private String name;

	@OneToMany (mappedBy = "folder")
	private List<Document> documents;
	
	@ManyToOne
	@JoinColumn(name ="id_library")
	private Library library;
	
	@OneToOne
	private Description description;

}

File: Project1/Model/Folder.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;

namespace Project1.Model
{
    public class Folder : BaseModel
    {
        
        public string Name { get; set; }

        public int LibraryId { get; set; }
        public Library Library { get; set; }

        public IList<Document> Documents { get; set; }
    }
}
